libs/librrgraph/src/base/rr_graph_view.h

```diff
@@ -413,6 +419,20 @@ class RRGraphView {
         return node_storage_.edge_switch(id, iedge);
     }
 
+    inline short edge_switch(RREdgeId edge_id) const {
```

Comment:
I like this style of wrapping all the flyweight data in accessor functions.

Reply:
Check performance though, as we might do multiple flyweight lookups in sequence to get the components of delay ... if slower we might have to expose the flyweight instead (at least in hot loops).